File: src/Presentation/Mintada.Web/src/models/RulerCoinTypesResponseDto.ts
import type { RulerIssuerCoinTypeGroupDto } from './RulerIssuerCoinTypeGroupDto';

export type RulerCoinTypesResponseDto = {
  rulerId: number;
  rulerName: string;
  portraitUrl?: string | null;
  infoHtml?: string | null;
  hasMultipleIssuers: boolean;
  issuerGroups: RulerIssuerCoinTypeGroupDto[];
};

File: src/Presentation/Mintada.Api/Dtos/RulerCoinTypesResponseDto.cs
namespace Mintada.Api.Dtos;

public class RulerCoinTypesResponseDto
{
    public int RulerId { get; set; }
    public string RulerName { get; set; } = string.Empty;
    public string? PortraitUrl { get; set; }
    public string? InfoHtml { get; set; }
    public bool HasMultipleIssuers { get; set; }
    public List<RulerIssuerCoinTypeGroupDto> IssuerGroups { get; set; } = new();
}
